### PR TITLE
chore: release config

### PR DESCRIPTION
**Pull request by mohsen-deriv**
- setup circleci
- added Dockerfile
- nginx config from old repo added

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+  publish_to_pages_production:
+    description: 'Publish to cloudflare pages'
+    steps:
+      - run:
+          name: 'Publish to cloudflare pages (production)'
+          command: |
+            npm i wrangler@2.0.19
+            cd _site
+            # npx wrangler pages publish . --project-name=deriv-developers-portal-pages --branch=main
```
Comment (mohsen-deriv) [inline .circleci/config.yml:108]: tib requested to comment out this line.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,4 @@
+FROM nginx:alpine
+COPY ./build /usr/share/nginx/html
```
Comment (mohsen-deriv) [inline Dockerfile:2]: the build output is in `build` folder, so I had to update it.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+            git clone https://github.com/binary-com/devops-ci-scripts
+            cd devops-ci-scripts/k8s-build_tools
+            echo $CA_CRT | base64 --decode > ca.crt
+            ./release.sh deriv-com-api << parameters.k8s_version >>
+  notify_slack:
+    description: 'Notify slack'
+    steps:
+      - slack/status:
+          include_project_field: false
+          failure_message: 'Release failed for api.deriv.com with version *$(cat _site/version)*'
```
Comment (thibault-deriv) [inline .circleci/config.yml:86]: The version file is in `build/version`

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+          success_message: 'Release succeeded for api.deriv.com with version *$(cat _site/version)*'
+          webhook: ${SLACK_WEBHOOK}
+  publish_to_pages_staging:
+    description: 'Publish to cloudflare pages'
+    steps:
+      - run:
+          name: 'Publish to cloudflare pages (staging)'
+          command: |
+            npm i wrangler@2.0.19
+            cd _site
```
Comment (thibault-deriv) [inline .circleci/config.yml:96]: `cd build`

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+            echo "New staging website - http://staging.cf-pages-deriv-developers-portal.deriv.com"
+
+  publish_to_pages_production:
+    description: 'Publish to cloudflare pages'
+    steps:
+      - run:
+          name: 'Publish to cloudflare pages (production)'
+          command: |
+            npm i wrangler@2.0.19
+            cd _site
```
Comment (thibault-deriv) [inline .circleci/config.yml:107]: `cd build`

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+            cd devops-ci-scripts/k8s-build_tools
+            echo $CA_CRT | base64 --decode > ca.crt
+            ./release.sh deriv-com-api << parameters.k8s_version >>
+  notify_slack:
+    description: 'Notify slack'
+    steps:
+      - slack/status:
+          include_project_field: false
+          failure_message: 'Release failed for api.deriv.com with version *$(cat _site/version)*'
+          success_message: 'Release succeeded for api.deriv.com with version *$(cat _site/version)*'
```
Comment (thibault-deriv) [inline .circleci/config.yml:87]: The version is in `build/version`

File: Dockerfile (hunk 2 above)
Comment (thibault-deriv) [reply Dockerfile:2]: OK :+1:

File: .circleci/config.yml (hunk 3 above)
Comment (mohsen-deriv) [reply .circleci/config.yml]: thanks resolved :pray:

File: .circleci/config.yml (hunk 5 above)
Comment (mohsen-deriv) [reply .circleci/config.yml]: thanks resolved :pray:

File: .circleci/config.yml (hunk 6 above)
Comment (mohsen-deriv) [reply .circleci/config.yml]: thanks resolved :pray:

**Review by thibault-deriv — APPROVED** [review]

**Pull request merged**